src/coreclr/jit/ssabuilder.cpp

```diff
@@ -1114,7 +1114,7 @@ void SsaBuilder::BlockRenameVariables(BasicBlock* block)
     {
         for (GenTree* const tree : stmt->TreeList())
         {
-            if (tree->OperIsSsaDef())
+            if (tree->OperIsStore() || tree->OperIs(GT_CALL))
```

Comment:
In actuality the same problem exists for un-analyzable stores as well. But that is a pre-existing issue.

<details>
<summary>Reproduction</summary>

```cpp
[MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.AggressiveOptimization)]
public static bool Problem()
{
    [MethodImpl(MethodImplOptions.NoInlining)]
    static void Throw()
    {
        _intStatic = 1;
        throw new Exception();
    }

    _intStatic = 2;

    try
    {
        Throw();
        _intStatic = 2;
    }
    catch (Exception)
    {
        if (_intStatic == 2)
        {
            return true;
        }
    }

    return false;
}
```
</details>